pulp_smash/tests/pulp3/file/api_v3/test_repo_version.py

```diff
@@ -332,3 +333,166 @@ def test_all(self):
+        self.addCleanup(self.client.delete, self.repo['_href'])
+        self.create_repo_version(self.repo)
+        self.repo = self.client.get(self.repo['_href'])
+
+    def create_repo_version(self, repo, secs=0):
+        """Create a new repository version.
+
+        Create a new repository version for each content unit present in Pulp,
+        a delay in seconds can be added between every repository version
+        creation.
+        """
+        for content in self.contents:
+            self.client.post(
+                repo['_versions_href'],
+                {'add_content_units': [content['_href']]}
+            )
+            sleep(secs)
+
+    def test_filter_invalid_content(self):
+        """Filter repository version by invalid content."""
+        with self.assertRaises(HTTPError):
+            page = self.filter_repo_version(
+                self.repo,
+                {'content': utils.uuid4()}
+            )
+            self.assertEqual(len(page['results']), 0, page['results'])
+
+    def test_filter_valid_content(self):
+        """Filter repository versions by valid content."""
+        content_filter = {'content': choice(self.contents)['_href']}
+        repo_versions = self.filter_repo_version(
+            self.repo, content_filter)['results']
+        for repo_version in repo_versions:
+            self.assertIn(
+                self.client.get(content_filter['content']),
+                get_content(self.repo, repo_version['_href'])['results']
+            )
+
+    def test_filter_invalid_date(self):
+        """Filter repository version by invalid date."""
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+        self.create_repo_version(repo, randint(1, 3))
+        criteria = utils.uuid4()
+        date_filters = (
+            {'created': criteria},
+            {'created__gt': criteria, 'created__lt': criteria},
+            {'created__gte': criteria, 'created__lte': criteria},
+            {'created__range': '{},{}'.format(criteria, criteria)}
+        )
+        for date_filter in date_filters:
+            page = self.filter_repo_version(repo, date_filter)
+            self.assertEqual(len(page['results']), 0, page['results'])
+
+    def test_filter_valid_date(self):
+        """Filter repository version by a valid date."""
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+        self.create_repo_version(repo, randint(1, 3))
+        dates = self.get_repo_data(repo, 'created')
+        date_filters = (
+            [1, {'created': dates[0]}],
+            [len(dates) - 2,
+             {'created__gt': dates[0], 'created__lt':dates[-1]}],
+            [len(dates),
+             {'created__gte': dates[0], 'created__lte': dates[-1]}],
+            [2, {'created__range': '{},{}'.format(dates[0], dates[1])}],
+        )
+        for date_filter in date_filters:
```

Comment:
When iterating through pairs like this, one can use semantically meaningful variable names, like so:

```python
for params, num_results in ...:
```

This lets one avoid the semantically meaningless list indexing that's performed below.